core/base/common/MPIUtils.h

```diff
@@ -386,9 +386,10 @@ namespace ttk {
     }
     std::vector<int> sendVector(neighborSet.begin(), neighborSet.end());
     int localSize = neighborSet.size();
-    int sizes[ttk::MPIsize_];
-    int displacements[ttk::MPIsize_];
-    MPI_Gather(&localSize, 1, MPI_INT, sizes, 1, MPI_INT, 0, communicator);
+    std::vector<int> sizes(ttk::MPIsize_);
+    std::vector<int> displacements(ttk::MPIsize_);
```

Comment:
same as above

Reply:
`ttk::MPIsize_` is not a compile-time constant, we can't use `std::array` here.

Reply:
Is that a requirement?
See `core/base/progressiveTopology/ProgressiveTopology.h:121`

Reply:
It is a requirement indeed.

Reply:
`ttk::MPIsize` is a run-time variable set via `mpirun`. When enabling `-Wvla`, a warning appears on these lines. Replacing the C-array with a `std::array` (whose size should be a compile-time constant) returns a compiler error.